codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -782,11 +783,53 @@ picToObj' pic =
             return $ (unsafeCoerce arr :: JSVal)
     setProps xs obj = liftIO $ void $ mapM (\(s, v) -> setProp s v obj) xs
 
+-- x is the maximum display length (Int)
+-- y is the input string we want to trim
+-- if string (y) is less than x, then we return y
+-- otherwise, replace the interior of string with ...
+
+trim :: Int -> String -> String
+trim x y = case (x > (length y)) of
+      True -> y
+      False -> "..."
```

Comment:
For the case where `y` is too large, you probably still want to render part of the string.  The idea is that you could just chop out the middle, like this:

```
Vertices [1.0, 2.0, 3.0, 7...9.0, 1.0]
```

Reply:
Oh, so basically take from the first half of the string (that is longer than allotted string length), and the second half, accounting for the "..." length.

Reply:
@kammitama5: Yeah